scripts/msct_register.py

```diff
@@ -93,54 +116,96 @@ def register2d_centermassrot(fname_src, fname_dest, fname_warp='warp_forward.nii
     of the PCA. This function should be used on segmentations (not images).
     This works for 2D and 3D images.  If 3D, it splits the image and performs the rotation slice-by-slice.
     input:
-        fname_source: name of moving image (type: string)
-        fname_dest: name of fixed image (type: string)
+        fname_source: name of moving image (type: string), if rot > 1, this needs to be a list with the first element
+        being the image fname and the second the segmentation fname
+        fname_dest: name of fixed image (type: string), if rot > 1, needs to be a list
         fname_warp: name of output 3d forward warping field
         fname_warp_inv: name of output 3d inverse warping field
-        rot: estimate rotation with PCA (type: int)
+        rot: estimate rotation with PCA (=1), HOG (=2) or no rotation (=0) Default = 1
+        rot>1 needs image AND segmentation to work
         polydeg: degree of polynomial regularization along z for rotation angle (type: int). 0: no regularization
         verbose:
     output:
         none
     """
 
+    if rot > 1:  # following method wanting to use both im and seg could be tagged 3, 4, etc.
```

Comment:
why not using the boolean `im_and_seg ` you introduced earlier?

Reply:
Because it is not in that function (it is in `register_slicewise`, not `register2d_centermassrot`), that would imply to pass this parameter as an additional argument, which is not necessary given the fact that we are transmitting this info via the `rot` value

Reply:
ok, in that case i would make it explicit in the header of the function that rot=0 is no rot, rot=1 is rot method blabla, rot=2 is rot with method blublu

and then, in the code, be specific with the cases, i.e., instead of using rot>1, use rot ==1 or rot==2. Otherwise, in the future this might cause problem (e.g., a student implements rot=3, which does not need rotation, etc.)